tests/typecheck/test_request.yml

```diff
@@ -27,6 +27,14 @@
             reveal_type(request.user) # N: Revealed type is 'django.contrib.auth.models.User'
     custom_settings: |
         INSTALLED_APPS = ('django.contrib.contenttypes', 'django.contrib.auth')
+-   case: request_object_user_without_auth_and_contenttypes_apps
```

Comment:
What will happen if we have only `'django.contrib.contenttypes'` or only `'django.contrib.auth'`?

Reply:
Only having `django.contrib.auth` should not be supported, since Django docs explicitly list `django.contrib.contenttypes` as a dependency of the authentication framework. In case, someone gets an error related to that, it's their setup that's at fault.

> It’s generally a good idea to have the contenttypes framework installed; several of Django’s other bundled applications require it:
> The admin application uses it to log the history of each object added or changed through the admin interface.
Django’s authentication framework uses it to tie user permissions to specific models.

If they only have `django.contrib.contenttypes`, I'd expect this to work as is. I've added one more test for this case just to be sure.